cvxpylayers/torch/cvxpylayer.py

```diff
@@ -150,85 +149,79 @@ def forward(ctx, *params):
             params_numpy = [to_numpy(p) for p in params]
 
             # infer dtype, device, and whether or not params are batched
-            param_0_shape = params[0].shape
             ctx.dtype = params[0].dtype
             ctx.device = params[0].device
-            ctx.batch = len(param_0_shape) > len(param_order[0].shape)
 
-            # check dtype, device of params
-            for i, p in enumerate(params):
+            ctx.batch_sizes = []
+            for i, (p, q) in enumerate(zip(params, param_order)):
+                # check dtype, device of params
                 if p.dtype != ctx.dtype:
-                    raise RuntimeError(
+                    raise ValueError(
                         "Two or more parameters have different dtypes. "
                         "Expected parameter %d to have dtype %s but "
                         "got dtype %s." %
                         (i, str(ctx.dtype), str(p.dtype))
                     )
                 if p.device != ctx.device:
-                    raise RuntimeError(
+                    raise ValueError(
                         "Two or more parameters are on different devices. "
                         "Expected parameter %d to be on device %s "
                         "but got device %s." %
                         (i, str(ctx.device), str(p.device))
                     )
 
-            # check shapes of params
+                # check and extract the batch size for the parameter
+                # 0 means there is no batch dimension for this parameter
+                # and we assume the batch dimension is non-zero
+                if p.ndimension() == q.ndim:
+                    batch_size = 0
+                elif p.ndimension() == q.ndim + 1:
+                    batch_size = p.size(0)
+                    assert batch_size > 0
```

Comment:
sorry, just noticed this: what's the rationale behind this assertion? should this be a ValueError?

Reply:
Ah yeah, I'll update this to a ValueError